ibm/service/power/resource_ibm_pi_cloud_connection.go

```diff
@@ -497,3 +524,13 @@ func resourceIBMPICloudConnectionDelete(ctx context.Context, d *schema.ResourceD
 	d.SetId("")
 	return nil
 }
+
+func retryCloudConnectionsVPC(ccVPCRetry func() error, retryCount int, waitDuration time.Duration, operation string, errMsg error) (err error) {
+	for count := 0; count < retryCount; count++ {
+		log.Printf("[DEBUG] unable to get vpc details for cloud connection: %v", errMsg)
+		time.Sleep(waitDuration)
+		log.Printf("[DEBUG] retrying cloud connection %s, retry #%v", operation, count+1)
+		err = ccVPCRetry()
```

Comment:
This loop will keep on re-trying the func `retryCount` times. We should break if we get one successful call.

Reply:
You are using `errMsg != nil` in loop condition but it is never changed in the loop :)

Reply:
Try below code and test out with fake responses. I prefer that 2nd one.
```
func retryCloudConnectionsVPC(ccVPCRetry func() error, operation string, errMsg error) (err error) {
	for count := 0; count < vpcRetryCount && errMsg != nil; count++ {
		log.Printf("[DEBUG] unable to get vpc details for cloud connection: %v", errMsg)
		time.Sleep(vpcRetryDuration)
		log.Printf("[DEBUG] retrying cloud connection %s, retry #%v", operation, count+1)
		errMsg = ccVPCRetry()
	}
    err = errMsg
	return
}
```
OR
```
func retryCloudConnectionsVPC(ccVPCRetry func() error, operation string, errMsg error) error {
	for count := 0; count < vpcRetryCount && errMsg != nil; count++ {
		log.Printf("[DEBUG] unable to get vpc details for cloud connection: %v", errMsg)
		time.Sleep(vpcRetryDuration)
		log.Printf("[DEBUG] retrying cloud connection %s, retry #%v", operation, count+1)
		errMsg = ccVPCRetry()
	}
    return errMsg
}
```

Reply:
You can also pass errMsg pointer and not have any error return

Reply:
I just need whatever code is present should work fine and not affect the overall time. Also it should be well tested. If we add low quality code then people tend to copy paste same thing again and again.
We should do it right from the start. Cannot compensate for cloud performance but terraform should not put additional strain.